src/math/dd/dd_pdd.cpp

```diff
@@ -733,6 +743,8 @@ namespace dd {
                 rest = p;
             }
         }
+        e.m_lc = lc.root;
```

Comment:
factor_entry could be invalidated if there is a gc between cache lookup and the update here.
Instead of resetting the entire cache you can mark the entry as busy and avoid removing busy entries during gc.